src/OrchardCore/OrchardCore.DisplayManagement/Notify/NotifyFilter.cs

```diff
@@ -175,7 +179,7 @@ private string SerializeNotifyEntry(NotifyEntry[] notifyEntries)
             try
             {
                 var protector = _dataProtectionProvider.CreateProtector(nameof(NotifyFilter));
-                var signed = protector.Protect(JConvert.SerializeObject(notifyEntries, _settings));
```

Comment:
Commenting here as I cannot comment on existing code...

```
  _existingEntries = messageEntries.Concat(_existingEntries).Distinct(new NotifyEntryComparer(_htmlEncoder)).ToArray();
  ```
  
  Have I ever told that I don't like LINQ 😉 Could probably use HashSet directly and also ideally if possible, `NotifyEntryComparer.Default` (internal static readonly) if the serializer reference equals some known default, maybe `internal NotifyEntryComparer.Create()` could check such a thing and either create a new instance or use a default one